src/domain-engine.js

```diff
@@ -110,7 +110,8 @@ class DomainEngine {
         action.status !== snapshot.status
       ) {
         // Yes: recalculate state from the base
-        result = set(result, key, handler.call(source, base, action.payload))
+        var next = pipeline(steps, action.payload, base, scope)
```

Comment:
`pipeline()` function looks like it "acts on the steps" and returns the result. Maybe consider naming it something more explicit? `reduceSteps()`, `executeSteps`, etc. Also, before I looked at the `pipeline()` implementation, `next` as the result variable name had me thinking it only acted on the first step of the array. `result` is already expected/taken, maybe name it `stepsResult` ?

Reply:
Renamed `pipeline()` to `reduce()` and cut the `next` variable assignment in 000d18e